src/displayapp/screens/Calculator.cpp

```diff
@@ -0,0 +1,412 @@
+        min_width = 1;
+      }
+      while (tmp_offset < FIXED_POINT_OFFSET) {
+        tmp_offset *= 10;
+        min_width++;
+      }
+      min_width--;
+
+      for (uint8_t i = min_width; i < N_DECIMALS; i++) {
+        printRemainder /= 10;
+      }
+
+      if (offset == FIXED_POINT_OFFSET) {
+        lv_label_set_text_fmt(valueLabel, "%" PRId64, integer);
+      } else if ((offset == (FIXED_POINT_OFFSET / 10)) && (remainder == 0)) {
+        lv_label_set_text_fmt(valueLabel, "%" PRId64 ".", integer);
+      } else {
+        lv_label_set_text_fmt(valueLabel, "%" PRId64 ".%0*" PRId64, integer, min_width, printRemainder);
+      }
+    } break;
+  }
+}
+
+// update the result based on value and operation
+void Calculator::Eval() {
+  switch (operation) {
+    case ' ':
+      result = value;
+      break;
+
+    case '+':
+      // check for overflow
+      if (((result > 0) && (value > (MAX_VALUE - result))) || ((result < 0) && (value < (MIN_VALUE - result)))) {
+        error = Error::TooLarge;
+        break;
+      }
+
+      result += value;
+      break;
+    case '-':
+      // check for overflow
+      if (((result < 0) && (value > (MAX_VALUE + result))) || ((result > 0) && (value < (MIN_VALUE + result)))) {
+        error = Error::TooLarge;
+        break;
+      }
+
+      result -= value;
+      break;
+    case '*':
+      // check for overflow
+      // while dividing we eliminate the fixed point offset
+      // therefore we have to multiply it again for the comparison with value
+      if (((result != 0) && (value > (FIXED_POINT_OFFSET * (MAX_VALUE / result)))) ||
+          ((result != 0) && (value < (FIXED_POINT_OFFSET * (MIN_VALUE / result))))) {
+        error = Error::TooLarge;
+        break;
+      }
+
+      result *= value;
+      // fixed point offset was multiplied too
+      result /= FIXED_POINT_OFFSET;
+      break;
+    case '/':
+      // check for zero division
+      if (value == 0) {
+        error = Error::ZeroDivision;
+        break;
+      }
+
+      // fixed point offset will be divided too
+      result *= FIXED_POINT_OFFSET;
+      result /= value;
+      break;
+
+    // we use floats here because pow with fixed point numbers is weird
+    case '^': {
+      auto tmp_value = static_cast<double>(value);
+      tmp_value /= static_cast<double>(FIXED_POINT_OFFSET);
+
+      auto tmp_result = static_cast<double>(result);
+      tmp_result /= static_cast<double>(FIXED_POINT_OFFSET);
+
+      // check for overflow
+      // result^value > MAX_VALUE iff
+      // log2(result^value) > log2(MAX_VALUE) iff
+      // value * log2(result) > log2(MAX_VALUE)
+      if ((tmp_value * log2(std::abs(tmp_result))) > log2(static_cast<double>(MAX_VALUE))) {
+        error = Error::TooLarge;
+        break;
+      }
+
+      tmp_result = pow(tmp_result, tmp_value);
+      tmp_result *= static_cast<double>(FIXED_POINT_OFFSET);
+      result = static_cast<int64_t>(tmp_result);
+      break;
+    }
```

Comment:
This block of code is responsible for 70% of the increase in memory usage, probably due to the use of double. This must be addressed.

Reply:
Wow, that is unexpected for me. How did you find that out?
Could it also be from the `std::pow` library stuff?

But let's try to eliminate this. 1. We could try to implement the power function directly on fixed point numbers on our own. But that might take some time.
2. We could replace the power function with something else. But what? The easiest solution would be to make `=` or `0` larger for now and reserve the space. We could then eventually release the `^` button later.

Reply:
Because the FPU in the PineTime only supports 32-bit floats, doubles have to be calculated entirely in software, which would increase the code size a lot. Could you try using floats, and see if the precision is good enough? I personally think that floats are definitely precise enough for these calculations.

Reply:
Yes, I will try `floats`. But in theory the resolution should be too low.

Reply:
@Riksu9000, how can I measure how much memory this block takes?

Reply:
Delete it, recompile and see the difference in build size.

Reply:
Switching to floats only reduces size by 2788 bytes, which means that this block still takes more than 60% of the extra usage.

Reply:
Thank you for testing it. Then it's cmath, namely `pow` and `log2`?

Reply:
It is most likely those cmath functions. In which case, I think it's best to just remove the power function for now, because I haven't been able to find a suitable fixed-point power function yet. I propose moving the `=` to where the `^` is now, switching the `.` and `0`, and making the `0` twice as wide. This is how it's often done on keyboard numpads and calculators.

Reply:
I agree with cutting it out for now if the memory is too much.
@FintasticMan, you seem to have done it already. I like the layout. You can make a PR and I will merge it.

Regarding the other question: the doubles can stay?

Reply:
I've just made a PR against this branch.

> Regarding the other question: the doubles can stay?

I'm not entirely sure what you mean by this. The only doubles being used are the ones in the power operation, and those are being removed.

Reply:
You are right. I was not thinking while writing that question 😄

Reply:
I'm not entirely sure why the formatter is getting angry at the `buttonMap` now, when it didn't before...